src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java

```diff
@@ -0,0 +1,157 @@
+package de.tum.in.www1.artemis.config;
+
+import java.io.*;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.saml2.core.Saml2X509Credential;
+import org.springframework.security.saml2.provider.service.registration.InMemoryRelyingPartyRegistrationRepository;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrations;
+
+/**
+ * This class describes the security configuration for SAML2.
+ */
+// @formatter:off
+@Configuration
+@Order(1)
+@Profile("saml2")
+public class SAML2Configuration extends WebSecurityConfigurerAdapter {
+
+    private final Logger log = LoggerFactory.getLogger(SAML2Configuration.class);
+
+    private final SAML2Properties properties;
+
+    /**
+     * Constructs a new instance.
+     *
+     * @param properties The SAML2 properties
+     */
+    public SAML2Configuration(final SAML2Properties properties) {
+        this.properties = properties;
+
+        Security.addProvider(new BouncyCastleProvider());
```

Comment:
have you tested that this does not interfere with the use of SSH keys when the artemis server clones git repositories? We recently had issues here

Reply:
Currently, we do not use ssh clone. Could you explain possible issues in more details?

Reply:
The SSH library in JGit uses custom security providers. I think they try to use the first provider in Security, but I am not fully sure. In case the SSH library uses a different provider, it might fail for various reasons.

Could you activate SSH in one of your test systems to find out if it interferes with this SAML2 Configuration?